lib/matplotlib/dviread.py

```diff
@@ -898,50 +897,52 @@ def _parse(self, file):
-                            else:
-                                fontfile = word
+        if not line or line.startswith((b" ", b"%", b"*", b";", b"#")):
+            return
+        tfmname = basename = special = encodingfile = fontfile = None
+        matches = re.finditer(br'"([^"]*)(?:"|$)|(\S+)', line)
+        for match in matches:
+            quoted, unquoted = match.groups()
+            if unquoted:
+                if unquoted.startswith(b"<<"):  # font
+                    fontfile = unquoted[2:]
+                elif unquoted.startswith(b"<["):  # encoding
+                    encodingfile = unquoted[2:]
+                elif unquoted.startswith(b"<"):  # font or encoding
+                    if unquoted == b"<":
+                        word = next(filter(None, next(matches).groups()))
+                        if unquoted.endswith(b".enc"):
+                            encodingfile = word
+                        else:
+                            fontfile = word
+                    else:
+                        if unquoted.endswith(b".enc"):
+                            encodingfile = unquoted[1:]
                         else:
-                            if unquoted.endswith(b".enc"):
-                                encodingfile = unquoted[1:]
-                            else:
-                                fontfile = unquoted[1:]
-                    elif tfmname is None:
-                        tfmname = unquoted
-                    elif basename is None:
-                        basename = unquoted
-                elif quoted:
-                    special = quoted
-            if basename is None:
-                basename = tfmname
-            effects = {}
-            if special:
-                words = reversed(special.split())
-                for word in words:
-                    if word == b"SlantFont":
-                        effects["slant"] = float(next(words))
-                    elif word == b"ExtendFont":
-                        effects["extend"] = float(next(words))
-            self._font[tfmname] = PsFont(
-                texname=tfmname, psname=basename, effects=effects,
-                encoding=encodingfile, filename=fontfile)
+                            fontfile = unquoted[1:]
+                elif tfmname is None:
+                    tfmname = unquoted
+                elif basename is None:
+                    basename = unquoted
+            elif quoted:
+                special = quoted
+        if basename is None:
+            basename = tfmname
+        effects = {}
+        if special:
+            words = reversed(special.split())
+            for word in words:
+                if word == b"SlantFont":
+                    effects["slant"] = float(next(words))
+                elif word == b"ExtendFont":
+                    effects["extend"] = float(next(words))
+        if encodingfile is not None and not encodingfile.startswith(b"/"):
```

Comment:
This changes to caching the the file finding (it looks like this used to be done late and every time before).  I suspect this is OK and is an additional speed up, just checking that it was intentional.

Reply:
Yes (this doesn't change the public API, which is just that `__getitem__` returns resolved paths).  Note that `find_tex_file` itself has a `@lru_cache`, so the runtime cost doesn't really change here.

Reply:
ah, fair enough.  The lru_cache on find_tex_file was a detail not in my head so I retract my concerns.